test/utils/math/Math.test.js

```diff
@@ -354,20 +339,126 @@ describe('Math', function () {
   });
 
   describe('modExp', function () {
-    it('is correctly calculating modulus', async function () {
-      const base = 3n;
-      const exponent = 200n;
-      const modulus = 50n;
+    describe('with uint256 inputs', function () {
+      before(function () {
+        this.fn = '$modExp(uint256,uint256,uint256)';
+      });
+
+      it('is correctly calculating modulus', async function () {
+        const base = 3n;
+        const exponent = 200n;
+        const modulus = 50n;
+
+        expect(await this.mock[this.fn](base, exponent, modulus)).to.equal(base ** exponent % modulus);
+      });
+
+      it('is correctly reverting when modulus is zero', async function () {
+        const base = 3n;
+        const exponent = 200n;
+        const modulus = 0n;
+
+        await expect(this.mock[this.fn](base, exponent, modulus)).to.be.revertedWithPanic(PANIC_CODES.DIVISION_BY_ZERO);
+      });
+    });
+
+    describe('with bytes memory inputs', function () {
+      before(function () {
+        this.fn = '$modExp(bytes,bytes,bytes)';
+      });
+
+      it('is correctly calculating modulus', async function () {
+        const base = 3n;
+        const exponent = 200n;
+        const modulus = 50n;
+
+        expect(await this.mock[this.fn](toBeHex(base), toBeHex(exponent), toBeHex(modulus))).to.equal(
+          toBeHex(base ** exponent % modulus),
+        );
+      });
+
+      it('is correctly reverting when modulus is zero', async function () {
+        const base = 3n;
+        const exponent = 200n;
+        const modulus = 0n;
+
+        await expect(this.mock[this.fn](toBeHex(base), toBeHex(exponent), toBeHex(modulus))).to.be.revertedWithPanic(
+          PANIC_CODES.DIVISION_BY_ZERO,
+        );
+      });
 
-      expect(await this.mock.$modExp(base, exponent, modulus)).to.equal(base ** exponent % modulus);
+      for (const [baseExp, exponentExp, modulusExp] of product(range(0, 24, 4), range(0, 24, 4), range(0, 256, 64))) {
```

Comment:
These values are actually pretty small. The max value they take is
```
b: 1048576
e: 1048576
m: 6277101735386680763835789423207666416102355444464034512896
```
In comparaison, type(uint256).max is
```
115792089237316195423570985008687907853269984665640564039457584007913129639935
```

Reply:
wich makes me wonder:
- if all these fit in a uint256, why only test them by the bytes version, and not with the uint256 version
- This looks like fuzzing, but with very specific values (only powers of 2). 

Since the values used here are not bigger than then one supported by the foundry fuzzing, are way slower to run, and do not cover a bigger space ... do we need that at all ? I'd remove this part of the tests.